src/user/user.service.ts

```diff
@@ -233,23 +236,42 @@ export class UserService {
     return updatedUser;
   }
 
-  async resetPassword(id: number) {
-    const newPassword = uuidv4();
-    const hashedPassword = await this.passwordService.hash(newPassword);
+  async requestResetPassword(email: string) {
+    const user = await this.prisma.user.findUnique({ where: { email } });
+    if (!user) {
+      throw new BadRequestException(`User with email ${email} doesn't exists.`);
+    }
+    const verificationToken = this.authService.signEmail(user.email);
 
-    const user = await this.prisma.user.findUnique({ where: { id } });
     try {
-      await this.mailService.userPasswordReset(user, hashedPassword);
+      await this.mailService.requestUserPasswordReset(user, verificationToken);
     } catch {
       throw new InternalServerErrorException(
         "Failed to send 'password reset' email",
       );
     }
+  }
 
-    return await this.prisma.user.update({
-      where: { id },
-      data: { password: newPassword },
-    });
+  async resetPassword({ verificationToken, newPassword }: ResetPasswordDto) {
+    const email = this.authService.decodeEmail(verificationToken);
+    const user = await this.prisma.user.findUnique({ where: { email } });
+    if (!user) {
+      throw new BadRequestException(`User with email ${email} doesn't exists.`);
+    }
+
+    const hashedPassword = await this.passwordService.hash(newPassword);
+    try {
+      await this.prisma.user.update({
+        where: {
+          id: user.id,
+        },
+        data: {
+          password: hashedPassword,
+        },
+      });
+    } catch (e) {
+      throw e;
+    }
```

Comment:
Why do we even need try catch here if we just throw whatever error we get? At that point, just remove try and catch blocks as a whole.

Lets just remove them and we're good. Whatever error pops up will break the flow, which is fine